src/EventListener/StoreInDoctrineOnMessageSentListener.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace KaroIO\MessengerMonitorBundle\EventListener;
+
+use KaroIO\MessengerMonitorBundle\Storage\StoredMessage;
+use KaroIO\MessengerMonitorBundle\Storage\StoredMessageRepository;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Messenger\Event\SendMessageToTransportsEvent;
+
+/**
+ * @internal
+ */
+final class StoreInDoctrineOnMessageSentListener implements EventSubscriberInterface
+{
+    private $storedMessageRepository;
+
+    public function __construct(StoredMessageRepository $storedMessageRepository)
+    {
+        $this->storedMessageRepository = $storedMessageRepository;
+    }
+
+    public function onMessageSent(SendMessageToTransportsEvent $event): void
+    {
+        $this->storedMessageRepository->saveMessage(StoredMessage::fromEnvelope($event->getEnvelope()));
```

Comment:
I *am* ok with this `fromEnvelope` method, however - the whole point of `StoredMessage` is to be a representation of info from an `Envelope`.